File: 0206-reverse-linked-list/0206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode head ;
    public ListNode reverseList(ListNode head) {
        get(head) ; 
        return this.head ;
    }
    public ListNode get(ListNode head){
        if(head==null||head.next==null){
            this.head = head ;
            return head ; 
        }
        ListNode x = get(head.next); 
        x.next = head ;
        head.next = null ; 
        return head ;
    }
}

File: 0206-reverse-linked-list/0206-reverse-linked-list.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
#define Node struct ListNode* 
struct ListNode* reverseList(struct ListNode* head) {
    if(head==NULL){
        return NULL;
    }
    Node prev = NULL ;
    Node curr = head; 
    Node next = head->next ; 
    while(curr!=NULL){
        curr->next = prev ;
        prev = curr ; 
        curr = next ;
        if(next!=NULL)
        next = next->next ;
    }
    return prev ;
}